dbt/adapters/trino/connections.py

```diff
@@ -360,12 +361,24 @@ class TrinoConnectionManager(SQLConnectionManager):
     def exception_handler(self, sql):
         try:
             yield
-        # TODO: introspect into `DatabaseError`s and expose `errorName`,
-        # `errorType`, etc instead of stack traces full of garbage!
-        except Exception as exc:
-            logger.debug("Error while running:\n{}".format(sql))
-            logger.debug(exc)
-            raise dbt.exceptions.RuntimeException(str(exc))
+        except trino.exceptions.ProgrammingError as e:
+            msg = str(e)
+            if isinstance(e, trino.exceptions.TrinoQueryError):
+                logger.debug("Trino query id: {}".format(e.query_id))
+            logger.debug("Trino error: {}".format(msg))
+
+            raise DatabaseException(msg)
+        except Exception as e:
+            msg = str(e)
+            if "Failed to establish a new connection" in msg:
+                raise FailedToConnectException(msg) from e
+
+            if isinstance(e, RuntimeException):
```

Comment:
Is it possible to raise other exception than `RuntimeException`? Perhaps returning some default exception would be good to handle some exotic errors.